coriolis/conductor/rpc/server.py

```diff
@@ -344,7 +344,14 @@ def _check_reservation_for_transfer(self, transfer_action):
             LOG.debug(
                 "Attempting to check reservation with ID '%s' for transfer "
                 "action '%s'", reservation_id, action_id)
-            self._licensing_client.check_reservation(reservation_id)
+            try:
+                transfer_action.reservation_id = (
+                    self._licensing_client.check_refresh_reservation(
+                        reservation_id)['id'])
+            except Exception as ex:
+                if getattr(ex, 'code', None) in [404, 409]:
+                    self._check_create_reservation_for_transfer(
+                        transfer_action, transfer_type)
         else:
             LOG.debug(
```

Comment:
In this `else` block, please call `_check_create_reservation_for_transfer` in case we have appliances we will be adding licensing to later so that old replicas don't remain 'free to run'.